git/githistory/rewriter.go

```diff
@@ -314,12 +314,12 @@ func (r *Rewriter) rewriteTree(commitOID []byte, treeOID []byte, path string, fn
 		path := filepath.Join(path, entry.Name)
 
 		if !r.allows(entry.Type(), path) {
-			entries = append(entries, entry)
+			entries = append(entries, copyEntry(entry))
 			continue
 		}
 
 		if cached := r.uncacheEntry(entry); cached != nil {
-			entries = append(entries, cached)
+			entries = append(entries, copyEntry(cached))
```

Comment:
So we are creating a deep copy here. That looks correct to me. However, TBH I don't fully see why the shallow copy doesn't work. Are the entries used in any way *after* `r.db.WriteTree(rewritten)`? Until that point `tree.Entries` shouldn't change, no?

Reply:
Ahh. `range` [works on a copy](https://www.ardanlabs.com/blog/2017/06/for-range-semantics.html)?! That's why the pointers are not valid anymore when used in `rewritten`?

Reply:
https://github.com/golang/go/wiki/Range#gotchas

Reply:
That's the one 😄 .